src/main/java/com/hyperlocal/server/Utilities.java

```diff
@@ -50,37 +74,42 @@ public static String getPlaceHolderString(Integer numOfPlaceholders) {
       return result.toString();
     }
 
-    public static CompletableFuture<String> verifyAndDecodeIdJwt(String token) {
-      return getResponseBody(IDENTITY_API_JWKS_URL)
-          .thenApply((jsonWebKeySetString) -> {
-              try {
-                  // Set token and algorithm
-                  JsonWebSignature jsonWebSignature = new JsonWebSignature();
-                  jsonWebSignature.setAlgorithmConstraints(new AlgorithmConstraints(ConstraintType.PERMIT,   AlgorithmIdentifiers.RSA_USING_SHA256));
-                  jsonWebSignature.setCompactSerialization(token);
-                  
-                  // Find and use relevant JWK
-                  JsonWebKeySet jsonWebKeySet = new JsonWebKeySet(jsonWebKeySetString);
-                  JsonWebKey jsonWebKey = new VerificationJwkSelector().select(jsonWebSignature, jsonWebKeySet.getJsonWebKeys());
-                  jsonWebSignature.setKey(jsonWebKey.getKey());
+    public static String verifyAndDecodeIdJwt(String token) {
+        String jsonWebKeySetString =  getResponseBody(IDENTITY_API_JWKS_URL);
+        try {
+            // Set token and algorithm
+            JsonWebSignature jsonWebSignature = new JsonWebSignature();
+            jsonWebSignature.setAlgorithmConstraints(new AlgorithmConstraints(ConstraintType.PERMIT,   AlgorithmIdentifiers.RSA_USING_SHA256));
+            jsonWebSignature.setCompactSerialization(token);
+            
+            // Find and use relevant JWK
+            JsonWebKeySet jsonWebKeySet = new JsonWebKeySet(jsonWebKeySetString);
+            JsonWebKey jsonWebKey = new VerificationJwkSelector().select(jsonWebSignature, jsonWebKeySet.getJsonWebKeys());
+            jsonWebSignature.setKey(jsonWebKey.getKey());
 
-                  if(!jsonWebSignature.verifySignature()) return null;
+            if(!jsonWebSignature.verifySignature()) return null;
 
-                  return jsonWebSignature.getPayload();
-              } catch(Exception ex) {
-                  ex.printStackTrace();
-                  return null;
-              }
-          });
+            return jsonWebSignature.getPayload();
+        } catch(Exception ex) {
+            ex.printStackTrace();
+            return null;
+        }
     }
 
-    public static CompletableFuture<String> getResponseBody(String url) {
-      HttpClient client = HttpClient.newHttpClient();
-      HttpRequest request = HttpRequest.newBuilder()
-            .uri(URI.create(url))
-            .build();
-      return client.sendAsync(request, BodyHandlers.ofString())
-            .thenApply(HttpResponse::body);
+    public static String getResponseBody(String url) {
```

Comment:
We should not cache all the type of requests, this method needs rename and also make it private if possible so that it remains close to signature validation code

Reply:
Yep - that makes sense. Thanks!